test/test_treetensornetworks/test_solvers/test_contract_mpo.jl

```diff
@@ -51,4 +51,29 @@ using Test
   @test inner(psit, Hpsi) ≈ inner(psit, H, psi) atol = 1E-4
 end
 
+function asMPS(M::MPO, sites)
+  M_ = MPS(length(sites))
+  for n in eachindex(sites)
+    M_[n] = M[n]
+  end
+  return M_
+end
+
+@testset "Contract MPO-MPO" begin
+  nbit = 5
+  sites = siteinds("Qubit", nbit)
+  M1 = randomMPO(sites) + randomMPO(sites)
+  M2 = randomMPO(sites) + randomMPO(sites)
+
+  # The function `apply` does not work correctly with the mapping-MPO-to-MPS trick.
+  M1 = replaceprime(M1, 1=>2, 0=>1)
+
+  M2_ = asMPS(M2, sites)
+
+  M12_ref = asMPS(contract(M1, M2; alg="naive"), sites)
+  M12 = contract(M1, M2_; alg="fit")
```

Comment:
Isn't this only testing MPO-MPS contraction?

Reply:
Ah, I got your point. We are testing MPO-(MPS with two site indices) here. I'll rename it correctly.

Reply:
I guess I'm unclear about what the new functionality is. Is it MPO-MPO contraction, or a more general MPO-MPS contraction where the MPS can have dangling (uncontracted) indices?

Reply:
Originally, I intended to support `MPO-MPO` contraction. But, the latter case should work. I will add a test on the latter case.

Reply:
One goal of ITensorNetworks.jl is to make code "just work" whether it involves MPO or MPS, and in fact we may just not have separate types (that is still being debated). But it would be best if this kind of code didn't care much if the inputs are MPO or MPS (and as of #44, TTNO/TTNS types as well).